### PR TITLE
Build saas package

### PR DESCRIPTION
**Pull request by matthewelwell**
This PR updates the build process to bring in the standalone SAML module and include it in the source bundle before deploying to Elastic Beanstalk. 

As part of this work, I had to add a new file in the .ebextensions directory to upgrade pip before installing requirements, otherwise the pysaml2 install failed.

### CODE REVIEW

File: .github/workflows/api-deploy-production.yml [hunk 1]
```diff
@@ -22,6 +22,22 @@ jobs:
               with:
                   fetch-depth: 0
 
+            - name: Add SAML package
```
Comment (gagantrivedi) [inline .github/workflows/api-deploy-production.yml]: I have not explored much but, I think we can use Reusable workflows here? Put all of this in a common workflow and use that?

Comment (matthewelwell) [reply .github/workflows/api-deploy-production.yml]: Yep, good shout. 

They're kinda cool, but a bit annoying. 

The main issues I have: 

 * You seemingly have to provide a git reference for the workflow, you can't just use the workflow from the current commit (unless you manually set the branch). I tried using `${GITHUB_SHA}` and that just complained that it wasn't valid. I've just used `main` for now but we may want to switch it to a sha once this PR is merged. 
 * The way in which secrets work is a bit confusing. You have to provide the github environment through to the reusable job but then provide the secrets to the reusable workflow from the calling workflow. 

There's a chance we may be better off using local actions instead. I'll have a look into that quickly now but you can have a look at what's there in answer to your comment about reusable workflows at least.

Comment (gagantrivedi) [reply .github/workflows/api-deploy-production.yml]: > You seemingly have to provide a git reference for the workflow, you can't just use the workflow from the current commit 

That seems a bit annoying. 

>you can have a look at what's there in answer to your comment about reusable workflows at least 

Yep, that's what I was talking about

Comment (matthewelwell) [reply .github/workflows/api-deploy-production.yml]: Ok, I've implemented it using "local composite actions" instead. The documentation is pretty terrible but I think it's a bit neater. 

It solves the 2 issues I mentioned above but it comes with a small amount less code reuse: 

 * We have to checkout the code in the 'calling workflow'
 * We have to define the `container` & `runs-on` variables in the calling workflow

I still think it's a bit neater but I'm interested in your thoughts @gagantrivedi and @dabeeeenster

Comment (matthewelwell) [reply .github/workflows/api-deploy-production.yml]: For reference, here are the differences between the two options: https://github.com/Flagsmith/flagsmith/pull/667/commits/dfda7d640acbe538d4e63862bf7089f3103407ad

File: .github/actions/api-deploy/action.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        working-directory: api
+        env:
+          FLAGSMITH_SAML_REVISION: ${{ inputs.flagsmith_saml_revision }}
+        run: |
+          echo "Downloading and injecting flagsmith-saml module..."
+          TARBALL_URL="https://github.com/flagsmith/flagsmith-saml/tarball/${FLAGSMITH_SAML_REVISION}"
+          mkdir /tmp/flagsmith-saml
+          curl -L -u token:${{ inputs.github_access_token }} $TARBALL_URL | tar --strip-components=1 -xz -C /tmp/flagsmith-saml --
+          mv /tmp/flagsmith-saml/saml .
+        shell: bash
```
Comment (dabeeeenster) [inline .github/actions/api-deploy/action.yml:36]: Do we need these lines here?

File: .github/workflows/api-deploy-production.yml [hunk 3]
```diff
@@ -11,23 +11,22 @@ on:
 jobs:
     deploy-production:
         runs-on: ubuntu-latest
-        name: API Deploy to Production
```
Comment (dabeeeenster) [inline .github/workflows/api-deploy-production.yml]: Why is this removed?

File: .github/workflows/api-deploy-staging.yml [hunk 4]
```diff
@@ -11,7 +11,6 @@ on:
 jobs:
     deploy-staging:
         runs-on: ubuntu-latest
-        name: API Deploy to Staging
```
Comment (dabeeeenster) [inline .github/workflows/api-deploy-staging.yml]: and this?

File: .github/actions/api-deploy/action.yml (hunk 2 above)
Comment (matthewelwell) [reply .github/actions/api-deploy/action.yml:36]: Annoyingly, yes. It's a requirement for a custom Github action.

File: .github/workflows/api-deploy-production.yml (hunk 3 above)
Comment (matthewelwell) [reply .github/workflows/api-deploy-production.yml]: Re-added

File: .github/workflows/api-deploy-staging.yml (hunk 4 above)
Comment (matthewelwell) [reply .github/workflows/api-deploy-staging.yml]: Re-added

**Review by dabeeeenster — APPROVED** [review]

**Review by gagantrivedi — APPROVED** [review]
lgtm
